clustering/preprocess.py

```diff
@@ -19,8 +19,28 @@ def filter_restaurants(businesses, max_restaurants=None):
     (determined by looking at the review data).
     """
 
-    # TODO: filter the input DataFrame
-    return businesses
+    cols = ['business_id', 'is_open', 'review_count', 'categories']
+    businesses = businesses[cols]
+
+    businesses = businesses[businesses['review_count'] > 10]
+    businesses = businesses[businesses['is_open'] != 0]
+
+    valid_b = businesses.as_matrix()
```

Comment:
Can we do this filtering without converting to an intermediate matrix and going back? I'm not too familiar with DataFrames, but I suspect there's a way to do it, and it would be more efficient than building a matrix just to compute which indices to keep.